### PR TITLE
[time.zone.leap.overview] Fix example

### PR DESCRIPTION
**Pull request by hewillk**
`leap_seconds` can only be compared with `sys_time<Duration>`, so we are in nondeduced context here? 

Correct me if I'm wrong.

### CODE REVIEW

**Comment by tkoeppe** [comment]
Does the example compile?

**Comment by hewillk** [comment]
CC @HowardHinnant

**Comment by HowardHinnant** [comment]
The example as corrected compiles.  The uncorrected code does not.

**Comment by HowardHinnant** [comment]
nitpick:  I use `{}` for construction when possible to disambiguate it visually from a function call which uses `()`.

File: source/time.tex [hunk 1]
```diff
@@ -10133,7 +10133,7 @@
 \begin{example}
 \begin{codeblock}
 for (auto& l : get_tzdb().leap_seconds)
-  if (l <= 2018y/March/17d)
+  if (l <= sys_days(2018y/March/17d))
```
Comment (tkoeppe) [inline source/time.tex:10136]:
```suggestion
  if (l <= sys_days{2018y/March/17d})
```

**Review by jwakely — APPROVED** [review]
I don't think we need to follow that specification policy in examples, and this conversion is guaranteed to work without narrowing so the braces are fine.

**Pull request merged**